### PR TITLE
Updating Microsoft.Cci.Extensions to include the "new" keyword for applicable nested types

### PR DESCRIPTION
**Pull request by tannergooding**
CC. @ericstj, @Anipik

This should resolve https://github.com/dotnet/arcade/issues/1488

### CODE REVIEW

**Comment by tannergooding** [comment]
What is the best way to test this?

File: src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs [hunk 1]
```diff
@@ -494,6 +505,38 @@ public static IMethodDefinition GetHiddenBaseMethod(this IMethodDefinition metho
             return Dummy.Method;
         }
 
+        public static ITypeDefinition GetHiddenBaseType(this ITypeDefinition type, ICciFilter filter = null)
```
Comment (ericstj) [inline src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs]: Any reason not to bool here instead?

**Comment by ericstj** [comment]
Test through GenAPI similar to https://github.com/dotnet/arcade/pull/2526#issuecomment-483358985.

File: src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs (hunk 1 above)
Comment (tannergooding) [reply src/Microsoft.Cci.Extensions/Extensions/CSharp/CSharpCciExtensions.cs]: Was matching the existing signature for other `GetHiddenBase` methods

**Comment by tannergooding** [comment]
This is also updated files as expected and should be good to merge.

**Pull request merged**